Makefile

```diff
@@ -0,0 +1,53 @@
+#this will execute the program 
+OBJ=obj
+BIN=bin
+
+BINS=\
+	$(BIN)/hello_world_main \
+	$(BIN)/hello_world_main2 \
+	$(BIN)/hello_world_main3
+
+OBJS=\
+	$(OBJ)/hello-world.o
+
+all: make_dirs $(OBJS) $(BIN)
```

Comment:
can you check if below works, also try to understand why and how.

```
all: make_dirs $(BIN)
```

Reply:
sure